gdnative-core/src/core_types/vector2.rs

```diff
@@ -89,6 +90,17 @@ impl Vector2Godot for Vector2 {
         Vector2::new(self.y, -self.x)
     }
 
+    #[inline]
+    fn move_towards(self, to: Vector2, delta: f32) -> Self {
+        let vd = to - self;
+        let len = vd.length();
+        if len <= delta || approx::abs_diff_eq!(len, f32::EPSILON) {
+            to
+        } else {
+            Vector2::lerp(&self, to, delta / len)
```

Comment:
Maybe it makes sense to also add a branch to clamp to `self` when `delta` is negative, or just panic?

Reply:
it's good, this is result of test comparing gdscript and rust:
`gd_Test_7:   begining_vector: (50, 10), final_vector: (124.278984, 24.855797), detla: -75.75`
`rust_Test_7: begining_vector: (50,10),  final_vector: (124.278984,24.855797), detla: -75.75`

Reply:
I recognize that this is consistent with Godot, but I'm not really sure if this behavior is "defined" as per the method name and documentation. But this behavior does seem useful, so I guess it's OK for now.